src/blockchain/Blockchain.ts

```diff
@@ -44,7 +44,11 @@ export interface IBlockchainApi {
 export default class Blockchain implements IBlockchainApi {
   public static asArray(queryResult: QueryResult): any[] {
     const json =
-      queryResult && queryResult.encodedLength ? queryResult.toJSON() : null
+      queryResult &&
+      !queryResult.isEmpty &&
+      (queryResult instanceof Array ? !queryResult.every(e => e.isEmpty) : true)
```

Comment:
will not work as expected if any element is a bool (isEmpty => always false). Not sure how to proceed